### PR TITLE
Download profile info for each address once

### LINKED ISSUE #300: Wait to download CIP8 info for the same address
### What would you like?
When adding profiles using CIP8 from `stokado`, check if information for that address has been downloaded recently, and don't redownload if it has.

### Why is this needed?
If Valora is pulling in many transactions in the feed from the same address, it could be very slow to re-download the info so many times: https://github.com/celo-org/wallet/pull/37#discussion_r623258624

### PR DESCRIPTION
**Pull request by isabellewei**
### Description
 
When restoring an account, a large number of transactions is usually imported into the home feed. This PR ensures we are only downloading profile info for addresses we have received transactions from once, instead of downloading for every single transaction

### Other changes

Uploading profile info is no longer a fatal action, so the `try catch` in the import saga was also removed 

### Tested

restored an account to device

### How others should test

restore an account that has transaction history, watch logs to see it is only downloading profile info for each distinct address once

### Related issues

- Fixes #300 

### Backwards compatibility

yes

### CODE REVIEW

**Review by gnardini — COMMENTED** [review]
Looks really good! 😄  Left some style nits, thanks for working on this!

File: packages/mobile/src/transactions/saga.ts [hunk 1]
```diff
@@ -167,31 +167,33 @@ function* refreshRecentTxRecipients() {
...
-        return call(addProfile, trans)
-      }
-    })
+    Array.from(
+      new Set(
+        transactions
+          .filter(
+            (trans) => isTransferTransaction(trans) && trans.type === TokenTransactionType.Received
+          )
+          // @ts-ignore isTransferTransaction ensures that trans is of TransferItemFragment type and has address
```
[the remark this answers is not in the document]
Comment (isabellewei) [reply packages/mobile/src/transactions/saga.ts:193]: the transactions are `TransactionFeedFragment` and I can't figure out how to cast them as `TransferItemFragment` :/

**Review by gnardini — APPROVED** [review]
Looks great! I fixed a small broken test so that maybe we can have this included in 1.14 :)

**Comment by Celoqa** [comment]
Hi @isabellewei @jeanregisser We have performed this task in latest internal  build V1.14.1 & Test Flight build V1.14.0(59) and observed the followings.

- User is able to see the all transactions in activity section when user restore the account after couple of days.
Can you please guide us how we can download the transaction history to verify this history.
TIA!

**Pull request merged**